kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java

```diff
@@ -100,11 +104,29 @@
     private final List<ProcessGenerator> processGenerators = new ArrayList<>();
 
     public static ProcessCodegen ofCollectedResources(KogitoBuildContext context, Collection<CollectedResource> resources) {
+        HashMap<String, String> processSVGs = new HashMap<>();
+        boolean useSvgAddon = context.getAddonsConfig().useProcessSVG();
         List<Process> processes = resources.stream()
                 .map(CollectedResource::resource)
                 .flatMap(resource -> {
                     if (SUPPORTED_BPMN_EXTENSIONS.stream().anyMatch(resource.getSourcePath()::endsWith)) {
-                        return parseProcessFile(resource).stream();
+                        Collection<Process> p = parseProcessFile(resource);
+                        if (useSvgAddon && resource instanceof FileSystemResource) {
+                            File f = ((FileSystemResource) resource).getFile();
+                            String processFileCompleteName = f.getName();
+                            String fileName = processFileCompleteName.substring(0, processFileCompleteName.lastIndexOf("."));
+                            p.stream().forEach(process -> {
+                                resources.stream().filter(r -> r.resource().getSourcePath().endsWith(String.format(SVG_EXPORT_NAME_EXPRESION, fileName)))
+                                        .forEach(svg -> {
+                                            try {
+                                                processSVGs.put(process.getId(), new String(Files.readAllBytes(Paths.get(svg.resource().getSourcePath()))));
+                                            } catch (IOException e) {
+                                                LOGGER.error("\n IOException trying to add " + svg.resource().getSourcePath() + " with processId:" + process.getId() + "\n" + e.getMessage());
```

Comment:
add exception into logging, also suggest to break the build in case there is an error here, like https://github.com/kiegroup/kogito-runtimes/pull/1269/files#diff-808e4e8a6db0b4b8bd73c7a583e857cd996994b3290d222cb1ae71c5f2f08117R191
```suggestion
                                                LOGGER.error("\n IOException trying to add " + svg.resource().getSourcePath() + " with processId:" + process.getId() + "\n" + e.getMessage(), e);
```

Reply:
added!